mobile/apps/photos/lib/services/video_preview_service.dart

```diff
@@ -104,6 +197,56 @@ class VideoPreviewService {
     }
   }
 
+  Future<List<EnteFile>> _getFiles({
+    DateTime? beginDate,
+    bool onlyFilesWithLocalId = true,
+  }) async {
+    return await FilesDB.instance.getStreamingEligibleVideoFiles(
+      beginDate: beginDate,
+      userID: Configuration.instance.getUserID()!,
+      onlyFilesWithLocalId: onlyFilesWithLocalId,
+    );
+  }
+
+  Future<StreamingStatus> getStatus() async {
+    try {
+      // TODO: Should we consider all days we could have processed or last 60 days
+      await _ensurePreviewIdsInitialized();
+      final files = await _getFiles(
+        beginDate: null,
+        onlyFilesWithLocalId: false,
+      );
+      final Set<int> totalProcessed = fileDataService.previewIds.keys.toSet();
+      final Set<int> total = {};
+      final Set<int> processed = {};
+      int skipped = 0;
+
+      for (final file in files) {
+        if (totalProcessed.contains(file.uploadedFileID)) {
+          processed.add(file.uploadedFileID!);
+        } else if (file.pubMagicMetadata?.sv == 1) {
+          skipped++;
+          continue;
+        }
+        total.add(file.uploadedFileID!);
+      }
+
+      final double netProcessedItems =
+          total.isEmpty ? 0 : (processed.length / total.length).clamp(0, 1);
```

Comment:
I don't think this is going to give correct result. Can you please document at top what are you considering in status calculation as per product requirements?